### PR TITLE
Source file structure and added pie menu editor

### PR DESCRIPTION
**Pull request by ryjacky**
Basically adding a completely new editor page UI that could be used in the future for plugging in all the pie menu components, pie menu settings, etc.

Restructuring the source files to be more organized and refactored classes for a more precise name.

All files/folders inside red boxes with a cross means they're deprecated. Yellow underline means the file name is changed.

The following is the new file structure:
image

The following is the old file structure:
image

Basically I had been feeling lazy and putting multiple classes in one file which is clearly not a good practice and could cause confusions and now most of them have their own place with some of them are just better to be in the same file.

This probably is also better from the version control perspective because I will know what has changed just by looking at the filename of the changed files.

### CODE REVIEW

**Comment by dumbeau** [comment]
As always... you're still a legend, didn't know it could get cleaner!
